### PR TITLE
Pass FacesContext to ResourceVerifierFactory.getResourceVerifier() to demonstrate that ResourceVerifier should only be used in the context of a Request.

### PR DESCRIPTION
**Pull request by stiemannkj1**

### CODE REVIEW

**Comment by stiemannkj1** [comment]
@ngriffin7a, Please do a code review on this commit and then close this PR without merging it (in order to avoid breaking the build). Feel free to comment on the commit or send a PR to me with your review, whichever is easier.

File: src/main/java/com/liferay/faces/util/application/ResourceVerifierFactory.java [hunk 1]
```diff
@@ -31,21 +32,22 @@
 public abstract class ResourceVerifierFactory implements FacesWrapper<ResourceVerifierFactory> {
 
 	/**
-	 * @return  an instance of {@link ResourceVerifier} from the {@link ResourceVerifierFactory} found by the {@link
-	 *          FactoryExtensionFinder}.
+	 * Create (if necessary) and return an instance of {@link ResourceVerifier} from the {@link ResourceVerifierFactory}
+	 * found by the {@link FactoryExtensionFinder}.
```
Comment (ngriffin7a) [inline src/main/java/com/liferay/faces/util/application/ResourceVerifierFactory.java]: Please add the sentence: "The returned instance is designed to be created and used during execution of a request thread."

File: src/main/java/com/liferay/faces/util/application/ResourceVerifierFactory.java [hunk 2]
```diff
@@ -31,21 +32,22 @@
...
 				ResourceVerifierFactory.class);
 
-		return resourceVerifierFactory.getResourceVerifier();
+		return resourceVerifierFactory.getResourceVerifier(facesContext);
 	}
 
 	/**
-	 * Gets an instance of {@link ResourceVerifier} which is the first member of the delegation chain.
+	 * Create (if necessary) and return an instance of {@link ResourceVerifier} which is the first member of the
+	 * delegation chain.
```
Comment (ngriffin7a) [inline src/main/java/com/liferay/faces/util/application/ResourceVerifierFactory.java]: Please add the sentence: "The returned instance is designed to be created and used during execution of a request thread."

File: src/main/java/com/liferay/faces/util/render/internal/ResourceRendererUtilImpl.java [hunk 3]
```diff
@@ -48,9 +48,12 @@
 	// Logger
 	private static final Logger logger = LoggerFactory.getLogger(ResourceRendererUtilImpl.class);
 
+	// Private Constants
+	private static final String RESOURCE_VERIFIER_KEY = ResourceRendererUtilImpl.class.getName() + "_" +
+		"ResourceVerifier";
+
```
Comment (ngriffin7a) [inline src/main/java/com/liferay/faces/util/render/internal/ResourceRendererUtilImpl.java]: Although it is not a significant performance impact due to the static nature of this String constant, Shyuang said that for two or three strings, it is better to use String.concat(String) than the plus operator. Let's keep that as a pattern if possible.

File: src/main/java/com/liferay/faces/util/render/internal/ResourceRendererUtilImpl.java [hunk 4]
```diff
@@ -191,19 +174,27 @@ public void setTransient(boolean transientFlag) {
 		this.transientFlag = transientFlag;
 	}
 
-	private String getComponentValue(UIComponent componentResource) {
+	private void logResourceDependencySatisfiedDebugMessage(UIComponent uiComponent) {
+
```
Comment (ngriffin7a) [inline src/main/java/com/liferay/faces/util/render/internal/ResourceRendererUtilImpl.java]: Please rename this method to simply logDebug

**Comment by ngriffin7a** [comment]
@stiemannkj1 These changes are very good. Please make the changes I mentioned in the code comments and send another PR. Thanks!

**Pull request closed without merging**